farm/data_handler/processor.py

```diff
@@ -299,13 +299,25 @@ def _featurize_samples(self):
                     sample.features = self._sample_to_features(sample=sample)
                 except Exception as e:
                     logger.error(f"Could not convert this sample to features: \n {sample}")
+                    logger.error(f"Basket id: id_internal: {basket.id_internal}, id_external: {basket.id_external}")
                     logger.error(f"Error message: {e}")
 
     def _create_dataset(self, keep_baskets=False):
         features_flat = []
+        basket_to_remove = []
         for basket in self.baskets:
             for sample in basket.samples:
-                features_flat.extend(sample.features)
+                if sample.features is None:
+                    basket_to_remove.append(basket)
+                else:
+                    features_flat.extend(sample.features)
```

Comment:
If one set of sample.features is None, all sample.features from the same basket need to be thrown out otherwise we may have features which belong to a basket which has been removed. So we should probably first iterate over all samples in the basket to check if any have features = None. If there are, we remove all features. If not, then we can add all to features_flat

Reply:
thanks @brandenchan for the suggestions. I am going to close this PR and create a new one.